libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp

```diff
@@ -448,6 +448,9 @@ bool NavEKF2_core::setOrigin(const Location &loc)
     // define Earth rotation vector in the NED navigation frame at the origin
     calcEarthRateNED(earthRateNED, EKF_origin.lat);
     validOrigin = true;
+#if HAL_GCS_ENABLED
+    gcs().send_message(MSG_ORIGIN);
```

Comment:
we should not be sending from within the EKF code here. 
Why move it into the EKFs?

Reply:
> we should not be sending from within the EKF code here. Why move it into the EKFs?

@tridge because every time the home position is set, the vehicle also emits GPS_GLOBAL_ORIGIN with HOME_POSITION.
There is no change in origin but emitted from the vehicle anyway.
Is this the desired behavior?
Also in the current implementation, GPS_GLOBAL_ORIGIN is emitted when the vehicle has a valid home location, not when it has a valid origin (which happens before the home is set).